compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/ConstantValueResolver.java

```diff
@@ -83,6 +95,7 @@ public void resolve(List<BLangConstant> constants, PackageID packageID) {
         constants.forEach(constant -> constant.accept(this));
         constantMap.clear();
         constants.forEach(constant -> checkUniqueness(constant));
+        constants.forEach(constant -> updateSymbolType(constant));
```

Comment:
Can't we set the type when visiting the constant(in L95)? 
In L105(`this.currentConstSymbol.value = visitExpr(constant.expr);`) we set the value. IMO after doing that we can set the type of the constant expr. 

Another thing is we set the type of constant expr in the `SymbolEnter`. As an example for the following program, we set the type as `int` in the symbol enter and then updated that type to a finite type in here. Is there any reason to do that? If not we can directly set the type after resolving the value of constant expr. 
Example
```ballerina
const int x = 3;
```

Reply:
> Can't we set the type when visiting the constant(in L95)?

We can. But in L97, we are removing duplicate constants. We can do the type updating process after that step to reduce the cost.

> Another thing is we set the type of constant expr in the SymbolEnter. As an example for the following program, we set the type as int in the symbol enter and then updated that type to a finite type in here. Is there any reason to do that? 

Ah no. For this scenario, the `symbolEnter` will update the symbol type to a finite type (since it is a numeric literal rather than an expression). In the constantValueResolver, we are updating the types of the `constants` which couldn't be resolved at the `symbolEnter` phase. Anyway, I think we need to move the `constantValueResolving` logic into the `symbolEnter` to do this in the proper way.

Reply:
> > Can't we set the type when visiting the constant(in L95)?
> 
> We can. But in L97, we are removing duplicate constants. We can do the type updating process after that step to reduce the cost.

IMO, the chances of someone duplicating a constant definition (or any definition for that matter) are quite low. I think it'll help with readability to do this when visiting the constant itself. It may also help when resolving constant references since the type is already set.